docs/book/src/reference/webhook-for-core-types.md

```diff
@@ -7,47 +7,58 @@ There is an [example](https://github.com/kubernetes-sigs/controller-runtime/tree
 in controller-runtime.
 
 It is suggested to use kubebuilder to initialize a project, and then you can
-follow the steps below to add admission webhooks for core types.
+follow the steps below to add admission webhooks for core types. The example shows
+how to set up a mutating webhook following the controller-runtime's webhook builder.
 
-## Implement Your Handler
+## Implement Your Webhook
 
-You need to have your handler implements the
-[admission.Handler](https://pkg.go.dev/sigs.k8s.io/controller-runtime/pkg/webhook/admission?tab=doc#Handler)
+You need to have your webhook to implement the
+[admission.CustomDefaulter](https://pkg.go.dev/sigs.k8s.io/controller-runtime/pkg/webhook/admission?tab=doc#CustomDefaulter)
 interface.
 
 ```go
-type podAnnotator struct {
-	Client  client.Client
-	decoder *admission.Decoder
+type podAnnotator struct {}
+
+func (a *podAnnotator) setupWebhookWithManager(mgr ctrl.Manager) error {
+       return ctrl.NewWebhookManagedBy(mgr).
+               For(&corev1.Pod{}).
+               WithDefaulter(a).
+               Complete()
 }
 
-func (a *podAnnotator) Handle(ctx context.Context, req admission.Request) admission.Response {
-	pod := &corev1.Pod{}
-	err := a.decoder.Decode(req, pod)
-	if err != nil {
-		return admission.Errored(http.StatusBadRequest, err)
-	}
+func (a *podAnnotator) Default(ctx context.Context, obj runtime.Object) error {
+    log := logf.FromContext(ctx)
+    pod, ok := obj.(*corev1.Pod)
+    if !ok {
+        return fmt.Errorf("expected a Pod but got a %T", obj)
+    }
 
 	// mutate the fields in pod
+    if pod.Annotations == nil {
+        pod.Annotations = map[string]string{}
+    }
+    pod.Annotations["example-mutating-admission-webhook"] = "foo"
+
+    log.Info("Annotated pod")
 
-	marshaledPod, err := json.Marshal(pod)
-	if err != nil {
-		return admission.Errored(http.StatusInternalServerError, err)
-	}
-	return admission.PatchResponseFromRaw(req.Object.Raw, marshaledPod)
+	return nil
 }
 ```
 
 **Note**: in order to have controller-gen generate the webhook configuration for
 you, you need to add markers. For example,
-`// +kubebuilder:webhook:path=/mutate-v1-pod,mutating=true,failurePolicy=fail,groups="",resources=pods,verbs=create;update,versions=v1,name=mpod.kb.io`
+`// +kubebuilder:webhook:path=/mutate--v1-pod,mutating=true,failurePolicy=fail,groups="",resources=pods,verbs=create;update,versions=v1,name=mpod.kb.io`
+(for webhooks the `path` is of format `/mutate-<group>-<version>-<kind>`. Since this documentation focuses on the core types the group is an empty string).
```

Comment:
The group is not empty for all cases, for example `deployments` is from the group `apps`  example: https://github.com/kubernetes-sigs/kubebuilder/blob/32e0fdc918b9a21485d7b042833402a62d0480f4/docs/kubebuilder_annotation.md?plain=1#L72

You might want to provide the helper to know the group which is in: https://github.com/kubernetes-sigs/kubebuilder/blob/32e0fdc918b9a21485d7b042833402a62d0480f4/pkg/plugins/golang/options.go#L29-L52

Reply:
> The group is not empty for all cases, for example `deployments` is from the group `apps` example:

The group is empty for `core` and this is explicitly called out in the sentence as suggested by @erikgb.

Reply:
That is not true, for deploments the group should be apps 
Such as we use in other markers Example: ` // +rbac:groups=apps,resources=deployments,verbs=get;list;watch;delete `

Reply:
I think our disconnect is with "core types" vs types belonging to `core` API group. With the latter, `<group>` is left empty.

Reply:
Apologies for the confusion, I'll reword the text a bit.